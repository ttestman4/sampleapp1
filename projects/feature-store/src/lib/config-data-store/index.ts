export * from './config-data-store.module';
