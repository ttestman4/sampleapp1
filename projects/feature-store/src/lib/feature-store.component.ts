import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'lib-feature-store',
  template: `
    <p>
      feature-store works!
    </p>
  `,
  styles: []
})
export class FeatureStoreComponent implements OnInit {

  constructor() { }

  ngOnInit() {
  }

}
