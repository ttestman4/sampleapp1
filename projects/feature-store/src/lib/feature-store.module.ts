import { NgModule } from '@angular/core';
import { FeatureStoreComponent } from './feature-store.component';

@NgModule({
  declarations: [FeatureStoreComponent],
  imports: [
  ],
  exports: [FeatureStoreComponent]
})
export class FeatureStoreModule { }
