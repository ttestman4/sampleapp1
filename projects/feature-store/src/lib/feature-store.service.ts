import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class FeatureStoreService {

  constructor() { }
}
