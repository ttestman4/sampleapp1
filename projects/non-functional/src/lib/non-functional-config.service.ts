import { InjectionToken } from '@angular/core';
import { NonFunctionalConfig } from './non-functional.models';

/**
 * This is not a real service, but it looks like it from the outside.
 * It's just an InjectionToken used to import the config object,
 * provided from the outside
 */
export const NonFunctionalConfigService =
    new InjectionToken<NonFunctionalConfig>('NonFunctionalConfig');
