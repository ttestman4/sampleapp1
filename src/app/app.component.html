<!--The content below is only a placeholder and can be replaced.-->
<div style="text-align:center">
  <h1>
    Flight Search App
  </h1>
</div>
<spa-search></spa-search>
<spa-result></spa-result>
<router-outlet></router-outlet>
