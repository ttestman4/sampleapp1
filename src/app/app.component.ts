import { Component } from '@angular/core';
import { CustomLogger } from 'non-functional';

@Component({
  selector: 'spa-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  title = 'spa';

  constructor(private logger: CustomLogger) {
    this.logger.debug('Your log message goes here');
    this.logger.debug('Multiple', 'Argument', 'support');
  }
}
