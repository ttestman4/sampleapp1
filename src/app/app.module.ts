import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { CustomLoggerLevel, NonFunctionalModule } from 'non-functional';
import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { environment } from 'src/environments/environment';
import { ConfigDataStoreModule, FlightStoreModule } from 'feature-store';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    NonFunctionalModule.forRoot({
      serverLoggingUrl: '/api/logs',
      level: CustomLoggerLevel.DEBUG,
      serverLogLevel: CustomLoggerLevel.ERROR,
      disableConsoleLogging: false,
      debugStore: !environment.production,
    }),
    ConfigDataStoreModule.forRoot(),
    FlightStoreModule.forRoot()
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
