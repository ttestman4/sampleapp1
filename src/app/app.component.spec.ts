import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { async, fakeAsync, tick, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { RouterTestingModule } from '@angular/router/testing';

import { ConfigureFn, configureTests } from '../lib/testing';

import { AppComponent } from './app.component';
import { NonFunctionalModule, CustomLoggerLevel } from 'non-functional';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [AppComponent],
          imports: [
            NoopAnimationsModule,
            NonFunctionalModule.forRoot({
              serverLoggingUrl: '/api/logs',
              level: CustomLoggerLevel.DEBUG,
              serverLogLevel: CustomLoggerLevel.ERROR,
              disableConsoleLogging: false,
            })
          ],
          schemas: [NO_ERRORS_SCHEMA],
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(AppComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  it('should create the app',
    async(() => {
      const app = component;
      expect(app).toBeTruthy();
    })
  );

  it('snaps', () => {
    expect(fixture).toMatchSnapshot();
  });

  it(`should have as title 'spa'`,
    async(() => {
      const app = component;
      expect(app.title).toEqual('spa');
    })
  );

  it(
    'should render title in a h1 tag',
    async(() => {
      const compiled = fixture.debugElement.nativeElement;
      expect(compiled.querySelector('h1').textContent).toContain('Welcome to spa!');
    })
  );
});
