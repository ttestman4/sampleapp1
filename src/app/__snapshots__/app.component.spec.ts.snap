// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snaps 1`] = `
<spa-root
  logger={[Function NGXLogger]}
  title={[Function String]}
>
  <div
    style="text-align:center"
  >
    <h1>
       Flight Search App 
    </h1>
  </div><nfl-async-state-indicator
    _nghost-a-c3=""
    indicator="bar"
    ng-reflect-indicator="bar"
  >
    
  </nfl-async-state-indicator><spa-search /><spa-result /><router-outlet />
</spa-root>
`;
